lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -327,6 +337,33 @@ def type_codes
                             end
           end
         end
+
+        class GenericStatusWithLookup < GenericStatus
+
+        protected
+
+          def check_expected_status(test_response, expected)
+            test_response.send("#{response_method(expected)}?")
+          end
+
+        private
+
+          # @api private
+          # Convert a symbol representing the expected response class into
+          #   the name of the test method to call on the response object
+          #
+          # @param response_symbol [Symbol] representing the expected http
+          #   response class
+          # @return [String] name of the response method, minus the ending
+          #   question mark
+          def response_method(response_symbol)
+            {
+              success: 'successful',
```

Comment:
I'm wondering if we should match our naming conventions to Rails going forward

Reply:
I'm trying to make the minimum change to get over the deprecation warnings. I'm trying not to be progressive about the function -- new naming conventions and such. Unfortunately, before 5.2 does not have successful (is that right?) 5.2 and after squawks if you don't use it.